src/main/java/baritone/command/defaults/ToMapCommand.java

```diff
@@ -0,0 +1,223 @@
+                    BlockState bs = cr.getBlock(x, y, z);
+                    String id;
+                    if (bs == null) {
+                        id = "air";
+                    } else {
+                        try {
+                            Matcher m = p.matcher(bs.toString());
+                            if (m.find())
+                                id = m.group(1);
+                            else
+                                throw new Exception("Damn");
+                        } catch (Exception e) {
+                            System.out.println("no match found for " + bs.toString());
+                            logDirect("no match found for " + bs.toString());
+                            return;
+                        }
+                    }
+                    if (bs == null || !(bs.getBlock() instanceof AirBlock)) { // we must show this block
+                        topView.get(x).put(z, id);
+                        break;
+                    }
+                }
+                if (!topView.containsKey(x)) {
+                    if (!topView.get(x).containsKey(z)) {
+                        topView.get(x).put(z, "air");
+                    }
+                }
+            }
+        }
+        logDirect("Finished scanning");
+        int minX = Collections.min(topView.keySet());
+        int maxX = Collections.max(topView.keySet());
+        int minZ = Integer.MAX_VALUE;
+        int maxZ = Integer.MAX_VALUE;
+        for (Map<Integer, String> zId : topView.values()) {
+            int minZForX = Collections.min(zId.keySet());
+            if (minZForX < minZ)
+                minZ = minZForX;
+            int maxZForX = Collections.max(zId.keySet());
+            if (maxZForX < maxZ)
+                maxZ = maxZForX;
+        }
+        int offsetX = -minX;
+        int offsetZ = -minZ;
+        BufferedImage result = new BufferedImage(maxX + offsetX, maxZ + offsetZ, BufferedImage.TYPE_INT_ARGB);
+        for (int x = minX; x < maxX; x++) {
+            for (int z = minZ; z < maxZ; z++) {
+                String blockId = null;
+                try {
+                    blockId = topView.get(x).containsKey(z) ? topView.get(x).get(z) : "air";
+                    if (!blockId.equals("air")) {
+                        BufferedImage textureAtCoord = getTextureForBlockId(blockId);
+                        Color c = averageColor(textureAtCoord, blockId);
+                        result.setRGB(x, z, c.getRGB());
+                    } else {
+                        result.setRGB(x, z, Color.TRANSLUCENT);
+                    }
+                } catch (IOException e) {
+                    System.out.println("Error setting color for " + blockId);
+                    e.printStackTrace();
+                }
+            }
+        }
+        logDirect("Finished coloring");
+        try {
+            ImageIO.write(result, "png", new File("/home/nexor/output.png"));
```

Comment:
don't hardcode output directory
Better save it to screenshots or the world's baritone folder

Reply:
How?

Reply:
https://github.com/cabaletta/baritone/pull/2257#issuecomment-743911392